code/game/atoms.dm

```diff
@@ -684,13 +684,15 @@
 	SEND_SIGNAL(src, COMSIG_ATOM_EX_ACT, severity, target)
 
 /**
-  * React to a hit by a blob objecd
+  * React to a hit by a blob object. Returns whether or not to cancel further action
   *
   * default behaviour is to send the [COMSIG_ATOM_BLOB_ACT] signal
   */
 /atom/proc/blob_act(obj/structure/blob/B)
-	SEND_SIGNAL(src, COMSIG_ATOM_BLOB_ACT, B)
-	return
+	var/blob_act_result = SEND_SIGNAL(src, COMSIG_ATOM_BLOB_ACT, B)
+	if (blob_act_result & COMPONENT_CANCEL_BLOB_ACT)
+		return FALSE
+	return TRUE
```

Comment:
Any particular reason you aren't using `if(SEND_SIGNAL(args) & FLAG)`?

Reply:
Not really 🤷 

Make a \`\`\`suggestion comment if you want.